File: src/main/java/com/pubnub/api/core/PubnubException.java
package com.pubnub.api.core;

import com.fasterxml.jackson.databind.JsonNode;
import lombok.Builder;

@Builder
public class PubnubException extends Exception {
    private String errormsg = "";
    private PubnubError pubnubError = PubnubError.PNERROBJ_PUBNUB_ERROR;
    private JsonNode jso;
    private String response;
    private int statusCode;
}

File: pubnub-core/pubnub-core-api/src/commonMain/kotlin/com/pubnub/api/PubNubException.kt
package com.pubnub.api

/**
 * Custom exception wrapper for errors occurred during execution or processing of a PubNub API operation.
 *
 * @property errorMessage The error message received from the server, if any.
 * @property pubnubError The appropriate matching PubNub error.
 * @property jso The error json received from the server, if any.
 * @property statusCode HTTP status code.
 * @property affectedCall A reference to the affected call. Useful for calling [retry][Endpoint.retry].
 */
expect class PubNubException(pubnubError: PubNubError) : Exception {
    constructor(errorMessage: String?)
}